### PR TITLE
Feature PM-191: Balance Updating + uPort Balance Watcher

### PR DESCRIPTION
**Pull request by andre-meyer**
Added new action `refreshOlympiaTokenBalance`
Refactored `enableWatcher ` in integrations into `watchersEnabled` with string properties of which variables to watch. Empty array disables watchers.

### CODE REVIEW

File: src/integrations/baseIntegration.js [hunk 1]
```diff
@@ -9,15 +10,15 @@ class BaseIntegration {
...
-   * @param {number} opts.defaultTimeout - Time interval for watcher
+   * @param {string} opts.watchersEnabled - Array of watchable properties. Empty array disables watcher completely @see WATCHABLE
+   * @param {number} opts.watcherTimeout - Time interval for watcher
    */
-  constructor({ enableWatcher = true, defaultTimeout = 1000 } = {}) {
-    this.defaultTimeout = defaultTimeout
-    this.enableWatcher = enableWatcher
-    if (this.enableWatcher) {
-      this.watcherInterval = setInterval(this.watcher, defaultTimeout)
+  constructor({ watchersEnabled = values(WATCHABLE), watcherTimeout = 1000 } = {}) {
```
Comment (apanizo) [inline src/integrations/baseIntegration.js]: I think the communication between parent-children is getting to complex. Are you telling me for describing the behaviour I need to send an object with 2 variables, and one of them is an array? Too complex.

I think each child (_if we embrace the inheritance is something good in JS but that is another discussion_) should be responsible of what he subscribes and the parent should provide (in different methods all options to get subscribed).

The interval should be moved to each child. Something like:
```
//uport
 ...
 constructor() {

     setInterval(() => { 
        watchBalance()  <---- baseIntegration for each property should have a method
        watchAccount()
        // remote or metamask should contain more method properties
     }, uportTimeout)
 }

// remote provider same as uport but including more watchers
```

What are the benefits?
1. You do not have to store an array of constants
2. You do not neet  to send an object to their father
3. You avoid checks in the ```watch``` method for knowing if the child needs to check for a certain prop (like balance or network)
4. Easier to understand what each provider cares, and therefore easier to maintain and debug

Comment (andre-meyer) [reply src/integrations/baseIntegration.js]: Yes, that might be a good idea. I'll rewrite it

File: src/integrations/baseIntegration.js [hunk 2]
```diff
@@ -94,41 +79,28 @@ class BaseIntegration {
     throw new Error('Invalid Balance')
   }
 
-  /**
-   * Periodic updater to get all relevant information from this provider
-   * @async
-   */
+  watch = async (property, getter) => {
```
Comment (apanizo) [inline src/integrations/baseIntegration.js]: I suggest a few small changes:
> Add some documentation
```
/**
 * property - Any provider parameter like balance, account...
 * getter - Function to call for checking the previous property has been changed
**/
```
> SImplify the if statement and reduce nested code one level
```
const propUpdated = this[property] !== value
 if (!propUpdated) {
   return
 }
 const providerUpdate = { [property]: value }
 
 if (!this.walletEnabled) {
   providerUpdate.available = true
   this.walletEnabled = true
 }
  
 this.runProviderUpdate(this, providerUpdate)
```

**Comment by andre-meyer** [comment]
when backmerging, strip any uPort related fixes but keep the api changes to the integrations

**Pull request merged**